cluster/src/main/java/io/snappydata/gemxd/MemoryNotificationListener.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright (c) 2016 SnappyData, Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you
+ * may not use this file except in compliance with the License. You
+ * may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+ * implied. See the License for the specific language governing
+ * permissions and limitations under the License. See accompanying
+ * LICENSE file.
+ */
+package io.snappydata.gemxd;
+
+/**
+ * Created by sachin on 24/8/16.
+ */
+public interface MemoryNotificationListener {
+  public void notifyCriticalUp();
+  public void notifyEvictionUp();
+}
```

Comment:
Why a new interface MemoryNotificationListener? What's the issue with use ResourceListener directly (from wherever it is going to be)?